### ISSUE TITLE
Support creating all @QueryParam for multiple traits

### ISSUE THREAD

**Issue by ming2377** [mulesoft-labs/raml-for-jax-rs#278]
Currently, I have:
traits:
  **paging**:
    queryParameters:
      pageNumber:
        ...
      pageSize:
        ...
      limit:
        ...
  **partial**:
    queryParameters:
      fields:
        ...

/firmware:
  get:
    is: [traits.**paging**, traits.**partial**]

raml-to-jaxrs-maven-plugin generates followings:

GetFirmwareResponse getFirmware(@QueryParam("pageNumber") Integer pageNumber, @QueryParam("pageSize") Integer pageSize, @QueryParam("limit") Integer limit);

It's missing the @QueryParam("fields") String fields

**Comment by jpbelang** [comment]
This may be a bug in the parser.  I think the processing application (i.e. the parser) should add these.  I do nothing explicitly with traits. 

I've updated the version of the parser to 1.0.9 (the latest release).

@jstoiko should we send this over to the parser ?

**Comment by jstoiko** [comment]
@jpbelang: yes. where you able to reproduce? if so, let's [file an issue](https://github.com/raml-org/raml-java-parser/issues).

**Comment by jpbelang** [comment]
I'll build an example and file it.

**Comment by ming2377** [comment]
Sorry for the confustion, this only happens when I define the parameter as optional like:

traits:
  **paging**:
    queryParameters:
      pageNumber:
        ...
      pageSize:
        ...
      limit:
        ...
  **partial**:
    queryParameters:
      fields**?**:
        ...

**Comment by ming2377** [comment]
So I think this may not be a bug, however I feel even when a queryParameter is marked as optional like **fields?**, the generated method should still generate _@QueryParam("fields") string fields_
